movies/src/components/app.js

```diff
@@ -8,17 +8,30 @@ import Constants from './constants';
 
 export default function App() {
   const [moviesList, updateMovieList] = useState(Constants.MOVIES);
+  const [isMovieSelected, selectedMovieId] = useState(0);
 
-  const updateData = (movies) => {
+  const updateData = useCallback((movies) => {
     updateMovieList(movies);
+  }, [moviesList]);
+
+  const  selectMovieById = useCallback(()=> {
```

Comment:
this useCallback is not really needed: you create callback and call it immediately.
Use `useCallback` only when function is passed as a prop to other component:

Reply:
useCallback  has been removed